google-ads/src/test/functional/java/com/google/ads/googleads/lib/SmokeTest.java

```diff
@@ -76,22 +86,22 @@ public void ensureCanReadAllAvailableCustomersCampaigns() throws IOException {
     }
   }
 
-  private static void fetchAllCampaigns(
+  /** Fetches the {@code Customer} object for each accessible customer. */
+  private static void fetchAllCustomers(
       GoogleAdsClient client, List<String> accessibleCustomerIds) {
-    int numCampaigns = 0;
+    int numCustomers = 0;
     try (GoogleAdsServiceClient googleAdsServiceClient =
         client.getLatestVersion().createGoogleAdsServiceClient()) {
       for (String customerId : accessibleCustomerIds) {
         SearchPagedResponse response =
             googleAdsServiceClient.search(
                 customerId,
-                "SELECT campaign.id, campaign.status, campaign.start_date "
-                    + "FROM campaign "
-                    + "WHERE campaign.status != 'REMOVED'");
-        numCampaigns += StreamSupport.stream(response.iterateAll().spliterator(), false).count();
+                "SELECT customer.id, customer.manager, customer.currency_code,"
+                    + " customer.descriptive_name FROM customer");
+        numCustomers += StreamSupport.stream(response.iterateAll().spliterator(), false).count();
       }
     }
-    System.out.println("Retrieved " + numCampaigns + " campaigns");
-    assertTrue("Expected to read at least one campaign for smoke test.", numCampaigns > 0);
+    System.out.println("Retrieved " + numCustomers + " customers");
```

Comment:
I guess reading campaigns was slow?

Reply:
Not quite. It was that if the accessible accounts consists solely of manager accounts, then the query won't find any campaigns and the test will fail. I think querying `customer` is an equally valid test, since it still exercises the library on a `search` request.